ext/pg_text_decoder.c

```diff
@@ -392,6 +408,180 @@ pg_text_dec_from_base64(t_pg_coder *conv, char *val, int len, int tuple, int fie
 	return out_value;
 }
 
+#ifdef PG_TEXT_DECODER_TIMESTAMP_EXT
+static inline int char_to_digit(char c)
+{
+  return c - '0';
+}
+
+static int str4_to_int(const char *str)
+{
+  return char_to_digit(str[0]) * 1000
+       + char_to_digit(str[1]) * 100
+       + char_to_digit(str[2]) * 10
+       + char_to_digit(str[3]);
+}
+
+static int str2_to_int(const char *str)
+{
+  return char_to_digit(str[0]) * 10
+       + char_to_digit(str[1]);
+}
+
+static VALUE pg_text_decoder_timestamp_do(VALUE rstr, int with_timezone)
+{
+  const char *str = StringValuePtr(rstr);
```

Comment:
`StringValuePtr()` doesn't ensure, that the corresponding string is zero-terminated. Therefore `RSTRING_LEN()` should be checked as well.

Reply:
You`re right. Using the pg_define_coder function resolves this problem since the caller ensures that the data is zero terminated

Reply:
I just checked the `pg_coder_decode` function and it does *not* ensure the string is zero-terminated as the comment in the `pg_decoder.c` file states.
Either the `pg_coder_decode` function should ensure the string is zero-terminated or the comment in the `pg_decoder.c` file should be updated. The `pg_text_dec_float` function assumes the string is zero-terminated too.
What do you think is better, update the pg_coder_decode function or update the comment?